src/tsoding_bot_bot.erl

```diff
@@ -0,0 +1,24 @@
+-module(tsoding_bot_bot).
+-behaviour(gen_fsm).
+-export([start_link/0,
+         listen/2,
+         init/1,
+         terminate/3]).
+
+start_link() ->
+    gen_fsm:start_link(?MODULE, listen, [{debug, [trace]}]).
```

Comment:
I'm not sure if there is a ready-to-use mechanism for prod/dev use-case. We probably have to implement it ourselves. Moving to a separate issue.